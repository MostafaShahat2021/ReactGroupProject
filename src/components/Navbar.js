import React from 'react';
import { NavLink } from 'react-router-dom';

function Navbar() {
  return (
    <nav>
      <div className="logo">
        <image src="../images/logo.jpg" />
      </div>
      <ul className="navbar">
        <li className="navbar-brand">
          <h1>Space Travelers Hub</h1>
        </li>
        <li className="navbar-brand">
          <NavLink to="/">Rockets</NavLink>
        </li>
        <li className="navbar-brand">
          <NavLink to="/missions">Missions</NavLink>
        </li>
        <li className="navbar-brand">
          <NavLink to="/profile">My Profile</NavLink>
        </li>
      </ul>
    </nav>
  );
}

export default Navbar;
