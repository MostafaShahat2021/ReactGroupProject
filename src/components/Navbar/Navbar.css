nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #fff;
  padding: 5px 2em;
  border-bottom: 1px solid #ebe5e5;
  box-shadow: 0 0 5px #fff;
}

.nav-link, .active-link {
  color: #1b1beb;
}

.nav-link::after {
  content: ' ';
  display: block;
  width: 0;
  height: 1px;
  background-color: #1b1beb;
  transition: width 0.3s ease-in-out;
}

.active-link::after {
  content: ' ';
  margin-top: -8%;
  display: block;
  width: 100%;
  height: 1px;
  background-color: #1b1beb;
  transition: width 0.3s ease-in-out;
}

.nav-link:hover::after {
  width: 100%;
}

.v-line {
  background-color: #a39f9f;
  height: 15px;
  width: 2px;
  margin-top: 5px;
}

.logo-nav {
  display: flex;
  gap: 2em;
}

.logo img {
  width: 40px;
  height: 30px;
}

.navbar {
  display: flex;
  gap: 1em;
}
