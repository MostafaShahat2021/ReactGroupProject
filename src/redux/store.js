import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionSlice from './mission/missionSlice';

const store = configureStore({
  reducer: {
    mission: missionSlice,
    rockets: rocketsReducer,
  },
});

export default store;
