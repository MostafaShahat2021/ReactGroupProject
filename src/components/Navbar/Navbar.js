import React from 'react';
import { NavLink, useMatch } from 'react-router-dom';
import './Navbar.css';

function Navbar() {
  const isActive = useMatch({
    path: '/',
    // exact: true,
  });
  return (
    <nav>
      <div className="logo-nav">
        <div className="logo">
          <img alt="logo" src="Images/logo.jpg" />
        </div>
        <h1>Space Travelers Hub</h1>
      </div>
      <div>
        <ul className="navbar">
          <li className="navbar-brand">
            <NavLink
              // exact
              className={isActive ? ' active-link' : 'nav-link'}
              to="/"
            >
              Rockets
            </NavLink>
          </li>
          <br />
          <li className="navbar-brand">
            <NavLink
              className={useMatch('/missions') ? ' active-link' : 'nav-link'}
              to="/missions"
            >
              Missions
            </NavLink>
          </li>
          <li className="v-line" />
          <li className="navbar-brand">
            <NavLink
              className={useMatch('/profile') ? ' active-link' : 'nav-link'}
              to="/profile"
            >
              My Profile
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
}

export default Navbar;
