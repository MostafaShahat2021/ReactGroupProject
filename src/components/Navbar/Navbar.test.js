import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Navbar from './Navbar';

it('Rockets should render', () => {
  const rockets = render(
    <Provider store={store}>
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    </Provider>,
  );
  expect(rockets).toMatchSnapshot();
});
