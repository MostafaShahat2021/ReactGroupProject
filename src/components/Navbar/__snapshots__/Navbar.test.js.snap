// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <div
          class="logo-nav"
        >
          <div
            class="logo"
          >
            <img
              alt="logo"
              src="Images/logo.jpg"
            />
          </div>
          <h1>
            Space Travelers Hub
          </h1>
        </div>
        <div>
          <ul
            class="navbar"
          >
            <li
              class="navbar-brand"
            >
              <a
                aria-current="page"
                class=" active-link active"
                href="/"
              >
                Rockets
              </a>
            </li>
            <br />
            <li
              class="navbar-brand"
            >
              <a
                class="nav-link"
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li
              class="v-line"
            />
            <li
              class="navbar-brand"
            >
              <a
                class="nav-link"
                href="/profile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <div
        class="logo-nav"
      >
        <div
          class="logo"
        >
          <img
            alt="logo"
            src="Images/logo.jpg"
          />
        </div>
        <h1>
          Space Travelers Hub
        </h1>
      </div>
      <div>
        <ul
          class="navbar"
        >
          <li
            class="navbar-brand"
          >
            <a
              aria-current="page"
              class=" active-link active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <br />
          <li
            class="navbar-brand"
          >
            <a
              class="nav-link"
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li
            class="v-line"
          />
          <li
            class="navbar-brand"
          >
            <a
              class="nav-link"
              href="/profile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
